exporters/prometheus/src/main/java/io/opentelemetry/exporters/prometheus/PrometheusCollector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.prometheus;
+
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricProducer;
+import io.prometheus.client.Collector;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public final class PrometheusCollector extends Collector {
+  private final MetricProducer metricProducer;
+
+  private PrometheusCollector(MetricProducer metricProducer) {
+    this.metricProducer = metricProducer;
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    Collection<MetricData> allMetrics = metricProducer.getAllMetrics();
+    List<MetricFamilySamples> allSamples = new ArrayList<>(allMetrics.size());
+    for (MetricData metricData : allMetrics) {
+      allSamples.add(MetricAdapter.toMetricFamilySamples(metricData));
+    }
+    return allSamples;
+  }
+
+  /**
+   * Returns a new builder instance for this exporter.
+   *
+   * @return a new builder instance for this exporter.
+   */
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /** Builder utility for this exporter. */
+  public static class Builder {
+    private MetricProducer metricProducer;
+
+    /**
+     * Sets the metric producer for the collector. Required.
+     *
+     * @param metricProducer the {@link MetricProducer} to use.
+     * @return this builder's instance.
+     */
+    public Builder setMetricProducer(MetricProducer metricProducer) {
+      this.metricProducer = metricProducer;
```

Comment:
Since it is required, should this method check for not null? 
```suggestion
	  Utils.checkNotNull(metricProducer, "metricProducer");
      this.metricProducer = metricProducer;
```

Reply:
Done in the build to ensure that it was called before calling build (required semantic).